lib/compiler.js

```diff
@@ -670,51 +699,52 @@ function fixFileExtension(filepath, files, onlyScriptsExportingFunction) {
 
       if (!(otherFileExtension in getExcludedExtensions()) &&
         path.basename(f, otherFileExtension) == basename) {
-        if (!onlyScriptsExportingFunction)
-          results.push(otherFile);
-        else if (onlyScriptsExportingFunction &&
-          (typeof require.extensions[otherFileExtension]) === 'function') {
-          results.push(otherFile);
-        }
+        // if (!onlyScriptsExportingFunction)
```

Comment:
We don't keep commented-out code around, please remove.

If I understand this code correctly, the intention was:

 - return a file with any extension when `onlyScriptsExportingFunction` is not set
 - return a file with an extension that can be loaded by Node.js runtime (an extension that is registered in `require.extensions`) when `onlyScriptsExportingFunction` is set

We should preserve these two modes.

I am proposing to replace `onlyScriptsExportingFunction` arg with `scriptExtensions` and modify all function callers accordingly.

```js
// implementation here
if (!scriptExtensions || otherFileExtension in scriptExtensions)
  results.push(otherFile);

// usage - flag not set
fixFileExtension(filepath, files)

// usage -flag set
fixFileExtension(filepath, files, scriptExtensions)
```